### ISSUE TITLE
Shipment methods for backend order don't show on multistore setup

### ISSUE THREAD

**Issue by haelbichalex** [magento/magento2#21867]
### Preconditions (*)

1. Magento 2.3.0
2. Commerce and Community

### Steps to reproduce (*)

1. Create new Magento 2 Setup
2. Setup second website, store and store view
3. Enable **only** Flat Rate for first website
4. Enable **only** Free Shipping for second website
5. Create simple product and assign it to both websites
6. Create backend order for second shop and add the product there
7. Click on "Get shipping methods and rates"

### Expected result (*)

1. Only shows "Free Shipping"

### Actual result (*)

1. Only shows "Flat Rate"

The behavior itself was quite weird. I described one example here to make clear what I'm talking about, but there are some more examples for configurations which show the incorrect shipping methods.

Any help would be appreciated :-)
Regards

**Comment by magento-engcom-team** [comment]
Hi @haelbichalex. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@haelbichalex do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by haelbichalex** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @haelbichalex. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @haelbichalex, here is your [Magento instance](https://i-21867-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21867-2-3-develop.instances.magento-community.engineering/admin](https://i-21867-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by haelbichalex** [comment]
I was not able to reproduce this on the 2.3-develop system. Was this already fixed?

**Comment by haelbichalex** [comment]
@magento-engcom-team give me 2.3 instance

**Comment by ghost** [comment]
HI @haelbichalex thank you for you report, seems like this was fixed here -> https://github.com/magento/magento2/issues/16604

**Comment by haelbichalex** [comment]
@magento-engcom-team give me 2.3.0 instance

**Comment by magento-engcom-team** [comment]
Hi @haelbichalex. Thank you for your request. I'm working on Magento 2.3.0 instance for you

**Comment by magento-engcom-team** [comment]
Hi @haelbichalex, here is your [Magento instance](https://i-21867-2-3-0.instances.magento-community.engineering).
Admin access: [https://i-21867-2-3-0.instances.magento-community.engineering/admin](https://i-21867-2-3-0.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by haelbichalex** [comment]
I fixed this on my 2.3.0 setup by following the guide to apply composer patches here:
https://support.magento.com/hc/en-us/articles/360005484154-Create-a-patch-for-a-Magento-2-Composer-installation-from-a-GitHub-commit

I added two patches to my composer.json:

```json
"extra": {
    "magento-force": "override",
    "composer-exit-on-patch-failure": true,
    "patches": {
        "magento/module-quote": {
            "Fix issue getting rates while placing order from Admin panel - part 1": "patches/composer/patch1.diff",
            "Fix issue getting rates while placing order from Admin panel - part 2": "patches/composer/patch2.diff"
        }
    }
}
```
Those patches are:
1. b35b68130265163f2ed151ff94e0d9f7fc63d44c
2. cc00197d13d6244e32201e4c6bf00287e54747c5
